### ISSUE TITLE
feat: add Conda environment.yml support for typosquat detection

### ISSUE THREAD

**Issue by yunaremaia** [yunaremaia/taintrace#33]
## Summary

Conda is the dominant package manager in data science, ML, and scientific Python. `environment.yml` is the standard Conda environment specification that pins package versions. taintrace supports 19 lockfile formats (Cargo, npm, Poetry, pnpm, Yarn, Go, Ruby, PHP, Swift, Elixir, Pipfile, etc.) but has no parser for Conda environments.

## What

Add Conda `environment.yml` support to taintrace:
1. Parse `environment.yml` YAML format (extract `dependencies` list with `package=version` or `package` entries)
2. Register Conda packages in the known packages database (top 100+ Conda packages: numpy, pandas, scipy, scikit-learn, tensorflow, pytorch, matplotlib, etc.)
3. Enable typosquat detection for Conda dependencies

## Why

- Conda has a massive user base in data science and ML (10M+ users)
- AI agents frequently install Conda packages (`conda install numpy`)
- Typosquats in Conda are a real threat (e.g., `numpyy` instead of `numpy`)
- Fills a clear gap in taintrace's ecosystem coverage
- The `environment.yml` format is simple YAML — easy to parse

## Example

```yaml
# environment.yml
name: myproject
dependencies:
  - numpy=1.26.4
  - pandas>=2.0.0
  - scipy  # unpinned
  - numpyy=1.24.0  # typosquat!
```

taintrace should flag `numpyy` as HIGH/CRITICAL risk (similar to `numpy`).

## Acceptance criteria

- [ ] New parser method `_parse_conda_environment` in `src/taintrace/lockfile.py`
- [ ] Added `"environment.yml": ("conda", "_parse_conda_environment")` to `EXTENDED_FORMATS`
- [ ] Added `"conda"` to CLI `--ecosystem` choices
- [ ] Added 100+ Conda packages to `KnownPackagesDB._CONDA_PACKAGES`
- [ ] 10+ tests covering: valid environment.yml, typosquat detection, unpinned packages, missing file
- [ ] Updated README ecosystem table
- [ ] Updated `action.yml` to mention Conda support

## References

- Conda environment.yml spec: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually
- Top Conda packages: https://github.com/conda/conda/blob/main/conda/core/index.json (or anaconda.org top packages)
- Existing `_parse_pipfile_lock` as template: `src/taintrace/lockfile.py`

**Comment by yunaremaia** [comment]
Duplicata da #34 (mesmo título e conteúdo). Mantendo a #34 aberta com a label 'help wanted' já aplicada.
